ambry-network/src/main/java/com/github/ambry/network/http2/Http2BlockingChannel.java

```diff
@@ -68,118 +45,78 @@
-            .build());
-        pipeline.addLast(new Http2MultiplexHandler(new ChannelInboundHandlerAdapter()));
-      }
-    });
 
-    // Start the client.
-    channel = b.connect().syncUninterruptibly().channel();
-    logger.info("Connected to remote host");
-    http2StreamChannelBootstrap = new Http2StreamChannelBootstrap(channel).handler(new ChannelInitializer<Channel>() {
-      @Override
-      protected void initChannel(Channel ch) throws Exception {
-        ChannelPipeline p = ch.pipeline();
-        p.addLast(new Http2StreamFrameToHttpObjectCodec(false));
-        p.addLast(new HttpObjectAggregator(1024 * 1024));
-        p.addLast(new SimpleChannelInboundHandler<FullHttpResponse>() {
-          @Override
-          protected void channelRead0(ChannelHandlerContext ctx, FullHttpResponse msg) throws Exception {
-            Integer streamId = msg.headers().getInt(HttpConversionUtil.ExtensionHeaderNames.STREAM_ID.text());
-            if (streamId == null) {
-              logger.error("Http2ResponseHandler unexpected message received: " + msg);
-              return;
-            }
-            ctx.channel().attr(RESPONSE_PROMISE).getAndSet(null).setSuccess(msg.content().retainedDuplicate());
-            logger.trace("Stream response received.");
-          }
-        });
-      }
-    });
   }
 
   @Override
   public void disconnect() throws IOException {
-    channel.disconnect().syncUninterruptibly();
-    workerGroup.shutdownGracefully();
+
   }
 
+  /**
+   * A new stream channel is created for each send.
+   */
   @Override
   public void send(Send request) throws IOException {
-    ByteBufferChannel byteBufferChannel = new ByteBufferChannel(ByteBuffer.allocate((int) request.sizeInBytes()));
-    while (!request.isSendComplete()) {
-      request.writeTo(byteBufferChannel);
+    Channel streamChannel;
+    try {
+      streamChannel = http2MultiplexedChannelPool.acquire().get(500, TimeUnit.MILLISECONDS);
+    } catch (InterruptedException | ExecutionException | TimeoutException e) {
+      throw new IOException("Can't acquire stream channel: ", e);
     }
-    byteBufferChannel.getBuffer().position(0);
-    ByteBuf byteBuf = Unpooled.wrappedBuffer(byteBufferChannel.getBuffer());
 
-    Http2StreamChannel childChannel = http2StreamChannelBootstrap.open().syncUninterruptibly().getNow();
-    Http2Headers http2Headers = new DefaultHttp2Headers().method(HttpMethod.POST.asciiName()).scheme("https").path("/");
-    responsePromise = childChannel.eventLoop().newPromise();
-    childChannel.attr(RESPONSE_PROMISE).set(responsePromise);
+    ChannelPipeline p = streamChannel.pipeline();
+    p.addLast(new Http2StreamFrameToHttpObjectCodec(false));
+    p.addLast(new HttpObjectAggregator(http2MultiplexedChannelPool.getMaxContentLength()));
+    p.addLast(new SimpleChannelInboundHandler<FullHttpResponse>() {
+      @Override
+      protected void channelRead0(ChannelHandlerContext ctx, FullHttpResponse msg) throws Exception {
```

Comment:
I think we should also override exceptionCaught method and set and error to the promise. So we don't have to wait for 3 seconds if something is wrong.